ec2/elasticloadbalancer.py

```diff
@@ -0,0 +1,306 @@
+            'elb exists.')
+
+
+@operation
+def create_load_balancer(**_):
+
+    if ctx.node.properties['use_external_resource']:
+        return None
+
+    lb = _create_load_balancer()
+
+    health_checks = ctx.node.properties.get('health_check')
+
+    if isinstance(health_checks, list):
+        if health_checks:
+            _add_health_check_to_load_balancer(lb, health_checks)
+
+
+def _add_instance_to_load_balancer_list_in_properties(instance_id):
+
+    if 'instance_list' not in ctx.target.instance.runtime_properties.keys():
+        ctx.target.instance.runtime_properties['instance_list'] = []
+
+    ctx.target.instance.runtime_properties['instance_list'].append(instance_id)
+
+
+def _remove_instance_from_load_balancer_list_in_properties(instance_id):
+    if instance_id in ctx.target.instance.runtime_properties['instance_list']:
+        ctx.target.instance.runtime_properties['instance_list'].remove(
+            instance_id)
+
+
+@operation
+def remove_instance_from_load_balancer(**_):
+
+    ctx.logger.debug('Attempting to remove_instance_from_load_balancer.')
+
+    elb_name = \
+        utils.get_external_resource_id_or_raise(
+            'elb_name', ctx.target.instance)
+
+    instance_id = \
+        utils.get_external_resource_id_or_raise(
+            'instance_id', ctx.source.instance)
+
+    instance_list = [instance_id]
+
+    try:
+        lb = _get_existing_load_balancer(elb_name)
+        lb.deregister_instances(instance_list)
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Instance not removed from Load Balancer. '
+                                  '{0}'.format(str(e)))
+
+    ctx.logger.info(
+        'Instance {0} removed from Load Balancer {1}.'
+        .format(instance_id, elb_name))
+    _remove_instance_from_load_balancer_list_in_properties(instance_id)
+
+
+@operation
+def add_instance_to_load_balancer(**_):
+
+    ctx.logger.debug('Attempting to add_instance_to_load_balancer.')
+
+    elb_name = \
+        utils.get_external_resource_id_or_raise(
+            'elb_name', ctx.target.instance)
+
+    instance_id = \
+        utils.get_external_resource_id_or_raise(
+            'instance_id', ctx.source.instance)
+
+    try:
+        lb = _get_existing_load_balancer(elb_name)
+        lb.register_instances(instance_id)
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+            raise NonRecoverableError('Instance not added to Load Balancer. '
+                                      '{0}'.format(str(e)))
+
+    ctx.logger.info(
+        'Instance {0} added to Load Balancer {1}.'
+        .format(instance_id, elb_name))
+
+    _add_instance_to_load_balancer_list_in_properties(instance_id)
+
+
+def _add_health_check_to_load_balancer(load_balancer, health_check):
```

Comment:
in other places in the module you just use "lb". I think it's better to be consistent here.